### ISSUE TITLE
Error creating Transformer from CRS

### ISSUE THREAD

**Issue by slyoung20** [nmt28/C-SHELPh#11]
Hi, 

I'm receiving an error that states: Error creating Transformer from CRS 
My epsg number is 32651. 
I'm not sure what line of code is in error. 

Thank you, 
Sherry Young

**Comment by nmt28** [comment]
Hi,

Can you paste the full error?

what system are you using? Mac OS? Windows? Linux?

Can you let me know what h5 file you are using?

**Comment by slyoung20** [comment]
Good evening,

Thank you for your assistance. I really appreciate it. I'm on a windows system. I tried a few different ICESat files including ATL03_20230203162831_06891807_006_01.h5. The error was pertaining to line 17. It could be something I failed to change, I'm not the most experienced programmer.
The epsg number output from line 16 is:

epsg number: 32651

---------------------------------------------------------------------------

ProjError                                 Traceback (most recent call last)
Cell In[17], line 2
      1 # Orthometrically correct the data using the epsg code
----> 2 lat_utm, lon_utm, photon_h = cshelph.orthometric_correction(latitude, longitude, photon_h, epsg_code)

File ~\cshelph.py:83, in orthometric_correction(lat, lon, Z, epsg)
     81 def orthometric_correction(lat, lon, Z, epsg):
     82     # transform ellipsod (WGS84) height to orthometric height
---> 83     transformerh = Transformer.from_crs("epsg:4326", "epsg:3855", always_xy=True)
     84     X_egm08, Y_egm08, Z_egm08 = transformerh.transform(lon, lat, Z)
     86     # transform WGS84 proj to local UTM

File ~\AppData\Roaming\Python\Python310\site-packages\pyproj\transformer.py:625, in Transformer.from_crs(crs_from, crs_to, always_xy, area_of_interest, authority, accuracy, allow_ballpark, force_over, only_best)
    552 @staticmethod
    553 def from_crs(
    554     crs_from: Any,
   (...)
    562     only_best: Optional[bool] = None,
    563 ) -> "Transformer":
    564     """Make a Transformer from a :obj:`pyproj.crs.CRS` or input used to create one.
    565
    566     See:
   (...)
    623
    624     """
--> 625     return Transformer(
    626         TransformerFromCRS(
    627             cstrencode(CRS.from_user_input(crs_from).srs),
    628             cstrencode(CRS.from_user_input(crs_to).srs),
    629             always_xy=always_xy,
    630             area_of_interest=area_of_interest,
    631             authority=authority,
    632             accuracy=accuracy if accuracy is None else str(accuracy),
    633             allow_ballpark=allow_ballpark,
    634             force_over=force_over,
    635             only_best=only_best,
    636         )
    637     )

File ~\AppData\Roaming\Python\Python310\site-packages\pyproj\transformer.py:337, in Transformer.__init__(self, transformer_maker)
    331     raise ProjError(
    332         "Transformer must be initialized using: "
    333         "'from_crs' or 'from_pipeline'."
    334     )
    336 self._local = TransformerLocal()
--> 337 self._local.transformer = transformer_maker()
    338 self._transformer_maker = transformer_maker

File ~\AppData\Roaming\Python\Python310\site-packages\pyproj\transformer.py:105, in TransformerFromCRS.__call__(self)
     99 def __call__(self) -> _Transformer:
    100     """
    101     Returns
    102     -------
    103     _Transformer
    104     """
--> 105     return _Transformer.from_crs(
    106         self.crs_from,
    107         self.crs_to,
    108         always_xy=self.always_xy,
    109         area_of_interest=self.area_of_interest,
    110         authority=self.authority,
    111         accuracy=self.accuracy,
    112         allow_ballpark=self.allow_ballpark,
    113         force_over=self.force_over,
    114         only_best=self.only_best,
    115     )

File ~\AppData\Roaming\Python\Python310\site-packages\pyproj\_transformer.pyx:568, in pyproj._transformer._Transformer.from_crs()

File ~\AppData\Roaming\Python\Python310\site-packages\pyproj\_transformer.pyx:344, in pyproj._transformer.proj_create_crs_to_crs()

ProjError: Error creating Transformer from CRS.



Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Thanks I will look into this. Were you able to install all the required packages? 

conda install -c conda-forge geopandas utm numpy matplotlib s3fs xarray zarr pyproj proj-data h5py earthaccess h5netcdf dask

pip install cshelph

**Comment by slyoung20** [comment]
I was using a jupyter notebook. I installed each separately. When I tried that line of code in a terminal it wouldn't load.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
ok, I found your h5 file and works ok on my end, so this is likely a system or package issue.

Can you run:

`!conda list`

and let me know the output?

**Comment by slyoung20** [comment]
I started running "conda install -c conda-forge geopandas utm numpy matplotlib s3fs xarray zarr pyproj proj-data h5py earthaccess h5netcdf dask" again, and it's still running. I'll let you know once that is finished if it solves the problem. Thank you for your help. When using the jupyter notebook run_bathymetry_extraction.ipynb<http://localhost:8890/notebooks/run_bathymetry_extraction.ipynb> I should only have to change the file name to try the first run correct?

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Yes, only the input filename should need to be changed. I did take a quick look at your data. You may need to use the `start_lat` and `end_lat` parameters to zoom into your specific area, but we can come on to that later. 

For now, I suspect that it's either pyproj or proj-data that is causing the issue. You may need to set specific versions in the installation. Let me know when you can run `conda list`

**Comment by slyoung20** [comment]
Hi,

That was able to install and I was able to run conda list. Conda list is shown below:


# packages in environment at C:\ProgramData\anaconda3:
#
# Name                    Version                   Build  Channel
abseil-cpp                20211102.0           h0e60522_0    conda-forge
aiobotocore               2.5.0              pyhd8ed1ab_0    conda-forge
aiohttp                   3.8.4           py310h8d17308_1    conda-forge
aioitertools              0.11.0             pyhd8ed1ab_0    conda-forge
aiosignal                 1.3.1              pyhd8ed1ab_0    conda-forge
anaconda-client           1.11.2          py310haa95532_0
anaconda-navigator        2.4.1           py310haa95532_0
anaconda-project          0.11.1          py310haa95532_0
anyio                     3.5.0           py310haa95532_0
argon2-cffi               21.3.0             pyhd3eb1b0_0
argon2-cffi-bindings      21.2.0          py310h2bbff1b_0
arrow-cpp                 11.0.0          py310h7cee713_0
asciitree                 0.3.3                      py_2    conda-forge
asttokens                 2.0.5              pyhd3eb1b0_0
async-timeout             4.0.2              pyhd8ed1ab_0    conda-forge
attrs                     22.1.0          py310haa95532_0
aws-c-common              0.4.57               ha925a31_1
aws-c-event-stream        0.1.6                hd77b12b_5
aws-checksums             0.1.9                ha925a31_0
aws-sdk-cpp               1.8.185              hd77b12b_0
backcall                  0.2.0              pyhd3eb1b0_0
backports                 1.1                pyhd3eb1b0_0
backports.functools_lru_cache 1.6.4              pyhd3eb1b0_0
backports.tempfile        1.0                pyhd3eb1b0_1
backports.weakref         1.0.post1                  py_1
beautifulsoup4            4.12.2          py310haa95532_0
blas                      1.0                         mkl
bleach                    4.1.0              pyhd3eb1b0_0
blosc                     1.21.4               hdccc3a2_0    conda-forge
bokeh                     3.1.1              pyhd8ed1ab_0    conda-forge
boltons                   23.0.0          py310haa95532_0
boost-cpp                 1.78.0               h9f4b32c_3    conda-forge
botocore                  1.29.76            pyhd8ed1ab_0    conda-forge
bounded-pool-executor     0.0.3              pyhd8ed1ab_0    conda-forge
branca                    0.6.0              pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                hcfcfb64_8    conda-forge
brotli-bin                1.0.9                hcfcfb64_8    conda-forge
brotlipy                  0.7.0           py310h2bbff1b_1002
bzip2                     1.0.8                he774522_0
c-ares                    1.19.1               hcfcfb64_0    conda-forge
ca-certificates           2023.5.7             h56e8100_0    conda-forge
cached-property           1.5.2                hd8ed1ab_1    conda-forge
cached_property           1.5.2              pyha770c72_1    conda-forge
cairo                     1.16.0            hd694305_1014    conda-forge
certifi                   2023.5.7           pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py310h2bbff1b_3
cfitsio                   4.2.0                h9ebe7e4_0    conda-forge
chardet                   4.0.0           py310haa95532_1003
charset-normalizer        2.0.4              pyhd3eb1b0_0
click                     8.0.4           py310haa95532_0
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.7.2              pyhd8ed1ab_1    conda-forge
cloudpickle               2.2.1              pyhd8ed1ab_0    conda-forge
clyent                    1.2.2           py310haa95532_1
colorama                  0.4.6           py310haa95532_0
comm                      0.1.2           py310haa95532_0
conda                     23.5.0          py310h5588dad_1    conda-forge
conda-build               3.24.0          py310haa95532_0
conda-content-trust       0.1.3           py310haa95532_0
conda-pack                0.6.0              pyhd3eb1b0_0
conda-package-handling    2.0.2           py310haa95532_0
conda-package-streaming   0.7.0           py310haa95532_0
conda-repo-cli            1.0.41          py310haa95532_0
conda-token               0.4.0              pyhd3eb1b0_0
conda-verify              3.4.2                      py_1
console_shortcut          0.1.1                         4
contourpy                 1.1.0           py310h232114e_0    conda-forge
cryptography              39.0.1          py310h21b164f_0
curl                      7.86.0               heaf79c2_1    conda-forge
cycler                    0.11.0             pyhd8ed1ab_0    conda-forge
cytoolz                   0.12.0          py310h8d17308_1    conda-forge
dash                      2.10.2             pyhd8ed1ab_0    conda-forge
dask                      2023.6.0           pyhd8ed1ab_0    conda-forge
dask-core                 2023.6.0           pyhd8ed1ab_0    conda-forge
debugpy                   1.5.1           py310hd77b12b_0
decorator                 5.1.1              pyhd3eb1b0_0
defusedxml                0.7.1              pyhd3eb1b0_0
distributed               2023.6.0           pyhd8ed1ab_0    conda-forge
earthaccess               0.5.2              pyhd8ed1ab_0    conda-forge
entrypoints               0.4             py310haa95532_0
executing                 0.8.3              pyhd3eb1b0_0
expat                     2.5.0                h63175ca_1    conda-forge
fasteners                 0.17.3             pyhd8ed1ab_0    conda-forge
filelock                  3.9.0           py310haa95532_0
fiona                     1.8.22          py310h30f9a14_2    conda-forge
flask                     2.2.3              pyhd8ed1ab_0    conda-forge
folium                    0.14.0             pyhd8ed1ab_0    conda-forge
font-ttf-dejavu-sans-mono 2.37                 hab24e00_0    conda-forge
font-ttf-inconsolata      3.000                h77eed37_0    conda-forge
font-ttf-source-code-pro  2.038                h77eed37_0    conda-forge
font-ttf-ubuntu           0.83                 hab24e00_0    conda-forge
fontconfig                2.14.2               hbde0cde_0    conda-forge
fonts-conda-ecosystem     1                             0    conda-forge
fonts-conda-forge         1                             0    conda-forge
fonttools                 4.40.0          py310h8d17308_0    conda-forge
freetype                  2.12.1               ha860e81_0
freexl                    1.0.6                h67ca5e6_1    conda-forge
frozenlist                1.3.3           py310h8d17308_0    conda-forge
fsspec                    2023.6.0           pyh1a96a4e_0    conda-forge
future                    0.18.3          py310haa95532_0
gdal                      3.5.3           py310h198874d_5    conda-forge
geopandas                 0.13.2             pyhd8ed1ab_1    conda-forge
geopandas-base            0.13.2             pyha770c72_1    conda-forge
geos                      3.11.1               h1537add_0    conda-forge
geotiff                   1.7.1                h4ffd875_4    conda-forge
gettext                   0.21.1               h5728263_0    conda-forge
gflags                    2.2.2             ha925a31_1004    conda-forge
giflib                    5.2.1                h8cc25b3_3
glib                      2.76.3               h12be248_0    conda-forge
glib-tools                2.76.3               h12be248_0    conda-forge
glob2                     0.7                pyhd3eb1b0_0
glog                      0.5.0                h4797de2_0    conda-forge
gst-plugins-base          1.20.3               h001b923_2    conda-forge
gstreamer                 1.20.3               h6b5321d_2    conda-forge
h5netcdf                  1.2.0              pyhd8ed1ab_0    conda-forge
h5py                      3.8.0           nompi_py310hae7a582_100    conda-forge
hdf4                      4.2.15               h1b1b6ef_5    conda-forge
hdf5                      1.12.2          nompi_h2a0e4a3_100    conda-forge
icc_rt                    2022.1.0             h6049295_2
icu                       70.1                 h0e60522_0    conda-forge
idna                      3.4             py310haa95532_0
importlib-metadata        6.7.0              pyha770c72_0    conda-forge
importlib_metadata        6.7.0                hd8ed1ab_0    conda-forge
intel-openmp              2023.1.0         h59b6b97_46319
ipykernel                 6.19.2          py310h9909e9c_0
ipython                   8.12.0          py310haa95532_0
ipython_genutils          0.2.0              pyhd3eb1b0_1
itsdangerous              2.1.2              pyhd8ed1ab_0    conda-forge
jedi                      0.18.1          py310haa95532_1
jinja2                    3.1.2           py310haa95532_0
jmespath                  1.0.1              pyhd8ed1ab_0    conda-forge
joblib                    1.2.0              pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h2bbff1b_1
jsonpatch                 1.32               pyhd3eb1b0_0
jsonpointer               2.1                pyhd3eb1b0_0
jsonschema                4.17.3          py310haa95532_0
jupyter_client            8.1.0           py310haa95532_0
jupyter_core              5.3.0           py310haa95532_0
jupyter_events            0.6.3           py310haa95532_0
jupyter_server            2.5.0           py310haa95532_0
jupyter_server_terminals  0.4.4           py310haa95532_0
jupyterlab_pygments       0.1.2                      py_0
kealib                    1.5.0                h61be68b_0    conda-forge
kiwisolver                1.4.4           py310h232114e_1    conda-forge
krb5                      1.19.4               h5b6d351_0
lcms2                     2.14                 h90d422f_0    conda-forge
lerc                      3.0                  hd77b12b_0
libarchive                3.6.2                h2033e3e_1
libbrotlicommon           1.0.9                hcfcfb64_8    conda-forge
libbrotlidec              1.0.9                hcfcfb64_8    conda-forge
libbrotlienc              1.0.9                hcfcfb64_8    conda-forge
libclang                  14.0.6          default_hb5a9fac_1
libclang13                14.0.6          default_h8e68704_1
libcurl                   7.86.0               heaf79c2_1    conda-forge
libdeflate                1.17                 h2bbff1b_0
libexpat                  2.5.0                h63175ca_1    conda-forge
libffi                    3.4.4                hd77b12b_0
libgdal                   3.5.3                hc3442b3_5    conda-forge
libglib                   2.76.3               he8f3873_0    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
libkml                    1.3.0             hf2ab4e4_1015    conda-forge
liblief                   0.12.3               hd77b12b_0
libnetcdf                 4.8.1           nompi_h8c042bf_106    conda-forge
libogg                    1.3.5                h2bbff1b_1
libpng                    1.6.39               h8cc25b3_0
libpq                     15.1                 h04fd8be_1    conda-forge
libprotobuf               3.20.3               h12be248_0    conda-forge
librttopo                 1.1.0               he22b5cd_12    conda-forge
libsodium                 1.0.18               h62dcd97_0
libspatialindex           1.9.3                h39d44d4_4    conda-forge
libspatialite             5.0.1               h07bf483_22    conda-forge
libsqlite                 3.42.0               hcfcfb64_0    conda-forge
libssh2                   1.10.0               h680486a_3    conda-forge
libthrift                 0.15.0               h636ae23_1    conda-forge
libtiff                   4.4.0                h8a3f274_2
libvorbis                 1.3.7                he774522_0
libwebp                   1.2.4                hbc33d0d_1
libwebp-base              1.2.4                h2bbff1b_1
libxml2                   2.10.3               h0ad7f3c_0
libxslt                   1.1.37               h2bbff1b_0
libzip                    1.9.2                hfed4ece_1    conda-forge
libzlib                   1.2.13               hcfcfb64_5    conda-forge
locket                    1.0.0              pyhd8ed1ab_0    conda-forge
lxml                      4.9.2           py310h2bbff1b_0
lz4                       4.3.2           py310hbbb2075_0    conda-forge
lz4-c                     1.9.4                h2bbff1b_0
m2-msys2-runtime          2.5.0.17080.65c939c               3
m2-patch                  2.7.5                         2
mapclassify               2.5.0              pyhd8ed1ab_1    conda-forge
markupsafe                2.1.1           py310h2bbff1b_0
matplotlib                3.7.1           py310h5588dad_0    conda-forge
matplotlib-base           3.7.1           py310h51140c5_0    conda-forge
matplotlib-inline         0.1.6           py310haa95532_0
menuinst                  1.4.19          py310h59b6b97_0
mistune                   0.8.4           py310h2bbff1b_1000
mkl                       2023.1.0         h8bd8f75_46356
mkl-service               2.4.0           py310h2bbff1b_1
mkl_fft                   1.3.6           py310h4ed8f06_1
mkl_random                1.2.2           py310h4ed8f06_1
msgpack-python            1.0.5           py310h232114e_0    conda-forge
msys2-conda-epoch         20160418                      1
multidict                 6.0.4           py310h8d17308_0    conda-forge
multimethod               1.9.1              pyhd8ed1ab_0    conda-forge
munch                     3.0.0              pyhd8ed1ab_0    conda-forge
munkres                   1.1.4              pyh9f0ad1d_0    conda-forge
navigator-updater         0.4.0           py310haa95532_0
nbclassic                 0.5.5           py310haa95532_0
nbclient                  0.5.13          py310haa95532_0
nbconvert                 6.5.4           py310haa95532_0
nbformat                  5.7.0           py310haa95532_0
nest-asyncio              1.5.6           py310haa95532_0
networkx                  3.1                pyhd8ed1ab_0    conda-forge
notebook                  6.5.4           py310haa95532_0
notebook-shim             0.2.2           py310haa95532_0
numcodecs                 0.11.0          py310h00ffb61_1    conda-forge
numpy                     1.24.3          py310h055cbcc_1
numpy-base                1.24.3          py310h65a83cf_1
openjpeg                  2.5.0                hc9384bd_1    conda-forge
openssl                   1.1.1u               hcfcfb64_0    conda-forge
orc                       1.7.4                h623e30f_1
packaging                 23.0            py310haa95532_0
pandas                    2.0.2           py310h1c4a608_0    conda-forge
pandocfilters             1.5.0              pyhd3eb1b0_0
parso                     0.8.3              pyhd3eb1b0_0
partd                     1.4.0              pyhd8ed1ab_0    conda-forge
pathlib                   1.0.1              pyhd3eb1b0_1
pcre                      8.45                 hd77b12b_0
pcre2                     10.40                h17e33f8_0    conda-forge
pickleshare               0.7.5           pyhd3eb1b0_1003
pillow                    9.4.0           py310hd77b12b_0
pip                       23.1.2                   pypi_0    pypi
pixman                    0.40.0               h8ffe710_0    conda-forge
pkginfo                   1.9.6           py310haa95532_0
platformdirs              2.5.2           py310haa95532_0
plotly                    5.15.0             pyhd8ed1ab_0    conda-forge
pluggy                    1.0.0           py310haa95532_1
ply                       3.11            py310haa95532_0
pooch                     1.7.0              pyha770c72_3    conda-forge
poppler                   22.11.0              ha6c1112_0    conda-forge
poppler-data              0.4.12               hd8ed1ab_0    conda-forge
postgresql                15.1                 h392df6f_1    conda-forge
powershell_shortcut       0.0.1                         3
pqdm                      0.2.0              pyhd8ed1ab_0    conda-forge
proj                      9.1.0                h3863b3b_0    conda-forge
proj-data                 1.14                 hd8ed1ab_0    conda-forge
prometheus_client         0.14.1          py310haa95532_0
prompt-toolkit            3.0.36          py310haa95532_0
psutil                    5.9.0           py310h2bbff1b_0
pure_eval                 0.2.2              pyhd3eb1b0_0
py-lief                   0.12.3          py310hd77b12b_0
pyarrow                   11.0.0          py310h790e06d_0
pycosat                   0.6.4           py310h2bbff1b_0
pycparser                 2.21               pyhd3eb1b0_0
pygments                  2.15.1          py310haa95532_1
pyjwt                     2.4.0           py310haa95532_0
pyopenssl                 23.0.0          py310haa95532_0
pyparsing                 3.1.0              pyhd8ed1ab_0    conda-forge
pyproj                    3.4.1           py310h0e9dbc5_0    conda-forge
pyqt                      5.15.7          py310hd77b12b_0
pyqt5-sip                 12.11.0         py310hd77b12b_0
pyrsistent                0.18.0          py310h2bbff1b_0
pysocks                   1.7.1           py310haa95532_0
python                    3.10.9               h966fe2a_1
python-cmr                0.7.0              pyhd8ed1ab_0    conda-forge
python-dateutil           2.8.2              pyhd3eb1b0_0
python-fastjsonschema     2.16.2          py310haa95532_0
python-json-logger        2.0.7           py310haa95532_0
python-libarchive-c       2.9                pyhd3eb1b0_1
python-tzdata             2023.3             pyhd8ed1ab_0    conda-forge
python_abi                3.10                    2_cp310    conda-forge
pytz                      2022.7          py310haa95532_0
pywin32                   305             py310h2bbff1b_0
pywinpty                  2.0.10          py310h5da7b33_0
pyyaml                    6.0             py310h2bbff1b_1
pyzmq                     25.1.0          py310hd77b12b_0
qt-main                   5.15.6               hf0cf448_0    conda-forge
qt-webengine              5.15.9               hb9a9bb5_5
qtpy                      2.2.0           py310haa95532_0
qtwebkit                  5.212                h992fabe_7    conda-forge
re2                       2022.04.01           h0e60522_0    conda-forge
requests                  2.29.0          py310haa95532_0
requests-toolbelt         0.9.1              pyhd3eb1b0_0
rfc3339-validator         0.1.4           py310haa95532_0
rfc3986-validator         0.1.1           py310haa95532_0
rtree                     1.0.1           py310h1cbd46b_1    conda-forge
ruamel.yaml               0.17.21         py310h2bbff1b_0
ruamel.yaml.clib          0.2.6           py310h2bbff1b_1
ruamel_yaml               0.17.21         py310h2bbff1b_0
s3fs                      2023.6.0           pyhd8ed1ab_0    conda-forge
scikit-learn              1.2.2           py310hd266714_2    conda-forge
scipy                     1.10.1          py310h309d312_1
send2trash                1.8.0              pyhd3eb1b0_1
setuptools                65.6.3          py310haa95532_0
shapely                   2.0.1           py310h8c82734_0    conda-forge
sip                       6.6.2           py310hd77b12b_0
six                       1.16.0             pyhd3eb1b0_1
snappy                    1.1.10               hfb803bf_0    conda-forge
sniffio                   1.2.0           py310haa95532_1
sortedcontainers          2.4.0              pyhd8ed1ab_0    conda-forge
soupsieve                 2.4             py310haa95532_0
sqlite                    3.41.2               h2bbff1b_0
stack_data                0.2.0              pyhd3eb1b0_0
tbb                       2021.8.0             h59b6b97_0
tblib                     1.7.0              pyhd8ed1ab_0    conda-forge
tenacity                  8.2.2              pyhd8ed1ab_0    conda-forge
terminado                 0.17.1          py310haa95532_0
threadpoolctl             3.1.0              pyh8a188c0_0    conda-forge
tiledb                    2.11.3               h5689973_1    conda-forge
tinycss2                  1.2.1           py310haa95532_0
tinynetrc                 1.3.1              pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h2bbff1b_0
toml                      0.10.2             pyhd3eb1b0_0
tomli                     2.0.1           py310haa95532_0
toolz                     0.12.0          py310haa95532_0
tornado                   6.2             py310h2bbff1b_0
tqdm                      4.65.0          py310h9909e9c_0
traitlets                 5.7.1           py310haa95532_0
typing-extensions         4.6.3           py310haa95532_0
typing_extensions         4.6.3           py310haa95532_0
tzdata                    2023c                h04d1e81_0
ucrt                      10.0.22621.0         h57928b3_0    conda-forge
ujson                     5.4.0           py310hd77b12b_0
unicodedata2              15.0.0          py310h8d17308_0    conda-forge
urllib3                   1.26.16         py310haa95532_0
utf8proc                  2.6.1                h2bbff1b_0
utm                       0.7.0              pyhd8ed1ab_0    conda-forge
vc                        14.2                 h21ff451_1
vc14_runtime              14.34.31931         h5081d32_16    conda-forge
vs2015_runtime            14.34.31931         hed1258a_16    conda-forge
wcwidth                   0.2.5              pyhd3eb1b0_0
webencodings              0.5.1           py310haa95532_1
websocket-client          0.58.0          py310haa95532_4
werkzeug                  2.3.6              pyhd8ed1ab_0    conda-forge
wheel                     0.38.4          py310haa95532_0
win_inet_pton             1.1.0           py310haa95532_0
wincertstore              0.2             py310haa95532_2
winpty                    0.4.3                         4
wrapt                     1.15.0          py310h8d17308_0    conda-forge
xarray                    2023.5.0           pyhd8ed1ab_0    conda-forge
xerces-c                  3.2.4                h63175ca_2    conda-forge
xyzservices               2023.5.0           pyhd8ed1ab_1    conda-forge
xz                        5.4.2                h8cc25b3_0
yaml                      0.2.5                he774522_0
yarl                      1.9.2           py310h8d17308_0    conda-forge
zarr                      2.15.0             pyhd8ed1ab_0    conda-forge
zeromq                    4.3.4                hd77b12b_0
zict                      3.0.0              pyhd8ed1ab_0    conda-forge
zipp                      3.15.0             pyhd8ed1ab_0    conda-forge
zlib                      1.2.13               hcfcfb64_5    conda-forge
zstandard                 0.19.0          py310h2bbff1b_0
zstd                      1.5.5                hd43e919_0

Note: you may need to restart the kernel to use updated packages.

I restarted the kernel and tried to run again. Now I'm getting an error on Matplotlib.


ImportError                               Traceback (most recent call last)
Cell In[1], line 3
      1 import numpy as np
      2 import h5py as h5
----> 3 import matplotlib.pyplot as plt
      4 import pyproj
      5 from pyproj import Proj

File ~\AppData\Roaming\Python\Python310\site-packages\matplotlib\__init__.py:148
    144 from packaging.version import parse as parse_version
    146 # cbook must import matplotlib only within function
    147 # definitions, so it is safe to import from it here.
--> 148 from . import _api, _version, cbook, _docstring, rcsetup
    149 from matplotlib.cbook import sanitize_sequence
    150 from matplotlib._api import MatplotlibDeprecationWarning

File ~\AppData\Roaming\Python\Python310\site-packages\matplotlib\rcsetup.py:27
     25 from matplotlib import _api, cbook
     26 from matplotlib.cbook import ls_mapper
---> 27 from matplotlib.colors import Colormap, is_color_like
     28 from matplotlib._fontconfig_pattern import parse_fontconfig_pattern
     29 from matplotlib._enums import JoinStyle, CapStyle

File ~\AppData\Roaming\Python\Python310\site-packages\matplotlib\colors.py:51
     49 from numbers import Number
     50 import re
---> 51 from PIL import Image
     52 from PIL.PngImagePlugin import PngInfo
     54 import matplotlib as mpl

File C:\ProgramData\anaconda3\lib\site-packages\PIL\Image.py:103
     94 MAX_IMAGE_PIXELS = int(1024 * 1024 * 1024 // 4 // 3)
     97 try:
     98     # If the _imaging C module is not present, Pillow will not load.
     99     # Note that other modules should not refer to _imaging directly;
    100     # import Image and use the Image.core variable instead.
    101     # Also note that Image.core is not a publicly documented interface,
    102     # and should be considered private and subject to change.
--> 103     from . import _imaging as core
    105     if __version__ != getattr(core, "PILLOW_VERSION", None):
    106         msg = (
    107             "The _imaging extension was built for another version of Pillow or PIL:\n"
    108             f"Core version: {getattr(core, 'PILLOW_VERSION', None)}\n"
    109             f"Pillow version: {__version__}"
    110         )

ImportError: DLL load failed while importing _imaging: The specified module could not be found.



Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
ok, so this is a very busy environment so it's likely that there are competing packages and conflicts that are causing you issues.

Lets begin by creating a new clean environment:

`conda create -n bathy python=3.10`

This will create a new environment for you called "bathy". Then start this environment:

`conda activate bathy`

Then install mamba. This is like conda but much faster:

`conda install -c Conda-forge mamba`

Then install the packages we need:

`mamba install -c conda-forge geopandas utm numpy matplotlib s3fs xarray zarr pyproj proj-data h5py earthaccess h5netcdf dask tqdm`

Then:

`pip install cshelph`

In your notebook try and import the required packages and see if we can get that far

**Comment by slyoung20** [comment]
Good morning Dr. Thomas,

Thank you for your help. It appears that there is some issue still with imaging in the Pillow plugin in Jupyter. When checking in a python terminal it says:
Requirement already satisfied: Pillow in c:\users\18042\appdata\local\packages\pythonsoftwarefoundation.python.3.11_qbz5n2kfra8p0\localcache\local-packages\python311\site-packages (9.5.0)
The pillow site here says: https://pillow.readthedocs.io/en/stable/installation.html  Pillow and PIL cannot co-exist in the same environment. Before installing Pillow, please uninstall PIL. and Pillow >= 1.0 no longer supports import Image. Please use from PIL import Image instead. and Pillow >= 2.1.0 no longer supports import _imaging. Please use from PIL.Image import core as _imaging instead.

I've tried both of these, and get the following error still.


ImportError                               Traceback (most recent call last)
Cell In[8], line 1
----> 1 from PIL.Image import core as _imaging

File C:\ProgramData\anaconda3\lib\site-packages\PIL\Image.py:103
     94 MAX_IMAGE_PIXELS = int(1024 * 1024 * 1024 // 4 // 3)
     97 try:
     98     # If the _imaging C module is not present, Pillow will not load.
     99     # Note that other modules should not refer to _imaging directly;
    100     # import Image and use the Image.core variable instead.
    101     # Also note that Image.core is not a publicly documented interface,
    102     # and should be considered private and subject to change.
--> 103     from . import _imaging as core
    105     if __version__ != getattr(core, "PILLOW_VERSION", None):
    106         msg = (
    107             "The _imaging extension was built for another version of Pillow or PIL:\n"
    108             f"Core version: {getattr(core, 'PILLOW_VERSION', None)}\n"
    109             f"Pillow version: {__version__}"
    110         )

ImportError: DLL load failed while importing _imaging: The specified module could not be found.


I don't have enough programming experience to figure out how to point to the correct version and remove whatever needs to be removed in Jupyter to get past this error.


Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Hey, no problem. Environment issues are common but we can get it figured out.

When importing matplotlib are you in the new environment you created? You can access this using:

`conda activate Bathy`

It is possible that your python notebook environment is not finding the right environment.

When in the new python environment (bathy):

`mamba install -c condo-forge ipykernel`

After this, restart all of your python terminals and notebooks. You should then be able to do the following when in a notebook and select the Bathy environment:

<img width="399" alt="Screen Shot 2023-06-21 at 8 43 43 PM" src="https://github.com/nmt28/C-SHELPh/assets/22138389/ca3571c2-5d20-443f-b23a-dec7794ffd1b">

As far as I know, we don't need PIL/pillow so lets try and avoid it for now and come back to it if it gives us matplotlib import errors.

Lastly, when in the bath env do:

`conda list`

and paste the results here. :)

**Comment by slyoung20** [comment]
Good afternoon,

It looks like we need to be working with python version less than 3.6, so 3.5 or earlier and pillow version 5.4 or earlier to fix the error I'm receiving with matplotlib. I've tried installing earlier versions of python into the new environment, but Jupyter keeps reverting back to python 3.11 (I just reinstalled navigator on this computer). Is there a way to remove pillow from the code when running it? I couldn't get mamba to install either, so I'm still working with conda. I can't seem to start a new kernel.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
hmmm, python 3. is old now and I am able to run all what I need in python 3.11. It's strange that mamba wouldn't install. Lets try some things step by step. In your base environment (the one that starts when you open anaconda) try:

`conda install -n base -c condo-forge mamba`

Then lets try making a new env like before (based on 3.10 in anticipation of some 3.11 issues):

`conda create -n bathy python=3.10`

Confirm that you can make this work by:

`conda activate bathy`

If you get an error that the environment already exists, try:

`conda deactivate`

`conda remove -n bathy --all`

**Comment by slyoung20** [comment]
This is what I got:

(base) C:\Users\18042>conda install -n base -c condo-forge mamba
Collecting package metadata (current_repodata.json): failed

UnavailableInvalidChannel: HTTP 404 NOT FOUND for channel condo-forge <https://conda.anaconda.org/condo-forge>

The channel is not accessible or is invalid.

You will need to adjust your conda configuration to proceed.
Use `conda config --show channels` to view your configuration's current state,
and use `conda config --show-sources` to view config file locations.



(base) C:\Users\18042>conda config --show  channels
channels:
  - defaults

(base) C:\Users\18042>

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
ah apologies, conda-forge got corrected to condo-forge. Please try: `condo-forge`

**Comment by slyoung20** [comment]
Yes, I believe that's what I entered. Just tried again.

(base) C:\Users\18042>conda install -n base -c condo-forge mamba
Collecting package metadata (current_repodata.json): failed

UnavailableInvalidChannel: HTTP 404 NOT FOUND for channel condo-forge <https://conda.anaconda.org/condo-forge>

The channel is not accessible or is invalid.

You will need to adjust your conda configuration to proceed.
Use `conda config --show channels` to view your configuration's current state,
and use `conda config --show-sources` to view config file locations.


Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
looks like you're still passing in condo-forge? Try:

`conda install -n base -c conda-forge mamba`

**Comment by slyoung20** [comment]
Sorry, read that backwards. It's trying to install now. This is what it shows so far, this is what happened before and it never got past this.
(base) C:\Users\18042>conda install -n base -c conda-forge mamba
Collecting package metadata (current_repodata.json): done
Solving environment: unsuccessful initial attempt using frozen solve. Retrying with flexible solve.
Solving environment: unsuccessful attempt using repodata from current_repodata.json, retrying with next repodata source.
Collecting package metadata (repodata.json):

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
yes, conda installs can take a while which is why we want to use mamba instead. However, me need to use conda to first install mamba in a strange "chicken and egg" situation

If this takes a huge amount of time, you can try:

`conda clean --all`

before trying the install again

**Comment by slyoung20** [comment]
I cleared it and restarted it, but it still won't go past solving the environment.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
Good evening,

I've still gotten nowhere with this. When after I create a new environment, it still has all the packages installed. When I try to update matplotlib in the new environment this is what happens. I'm thinking this could be an issue with the version of navigator?

PS C:\Users\18042> conda update matplotlib
Collecting package metadata (current_repodata.json): done
Solving environment: \
The environment is inconsistent, please check the package plan carefully
The following packages are causing the inconsistency:

  - defaults/win-64::anaconda-client==1.11.2=py310haa95532_0
  - defaults/win-64::anaconda-navigator==2.4.2=py310haa95532_0
  - defaults/win-64::anaconda-project==0.11.1=py310haa95532_0
  - defaults/win-64::black==22.6.0=py310haa95532_0
  - defaults/win-64::conda-build==3.24.0=py310haa95532_0
  - defaults/win-64::conda-repo-cli==1.0.41=py310haa95532_0
  - defaults/win-64::holoviews==1.15.4=py310haa95532_0
  - defaults/win-64::huggingface_hub==0.10.1=py310haa95532_0
  - defaults/win-64::hvplot==0.8.2=py310haa95532_0
  - defaults/win-64::ipykernel==6.19.2=py310h9909e9c_0
  - defaults/noarch::ipywidgets==7.6.5=pyhd3eb1b0_1
  - defaults/win-64::jupyter==1.0.0=py310haa95532_8
  - defaults/win-64::jupyterlab==3.5.3=py310haa95532_0
  - defaults/win-64::jupyterlab_server==2.19.0=py310haa95532_0
  - defaults/win-64::jupyter_client==7.3.4=py310haa95532_0
  - defaults/win-64::jupyter_console==6.6.2=py310haa95532_0
  - defaults/win-64::jupyter_core==5.2.0=py310haa95532_0
  - defaults/win-64::jupyter_server==1.23.4=py310haa95532_0
  - defaults/win-64::nbclassic==0.5.2=py310haa95532_0
  - defaults/win-64::nbclient==0.5.13=py310haa95532_0
  - defaults/win-64::nbconvert==6.5.4=py310haa95532_0
  - defaults/win-64::nbformat==5.7.0=py310haa95532_0
  - defaults/win-64::notebook==6.5.4=py310haa95532_0
  - defaults/win-64::notebook-shim==0.2.2=py310haa95532_0
  - defaults/win-64::pylint==2.16.2=py310haa95532_0
  - defaults/noarch::pyls-spyder==0.4.0=pyhd3eb1b0_0
  - defaults/win-64::python-lsp-black==1.2.1=py310haa95532_0
  - defaults/win-64::python-lsp-server==1.7.1=py310haa95532_0
  - defaults/win-64::pytoolconfig==1.2.5=py310haa95532_1
  - defaults/win-64::qtconsole==5.4.0=py310haa95532_0
  - defaults/win-64::rope==1.7.0=py310haa95532_0
  - defaults/win-64::scrapy==2.8.0=py310haa95532_0
  - defaults/win-64::spyder==5.4.1=py310haa95532_0
  - defaults/win-64::spyder-kernels==2.4.1=py310haa95532_0
  - defaults/noarch::tldextract==3.2.0=pyhd3eb1b0_0
  - defaults/win-64::transformers==4.24.0=py310haa95532_0
  - defaults/win-64::widgetsnbextension==3.5.2=py310haa95532_0
done

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
Dr. Thomas,

Got it, finally. Thank you so much for your help. Evidentially, when you do a fresh install on Anaconda Navigator, it asks if you want to update it. If you do, it installs python 11, it will also do this if you try to set up a new environment in the terminal. If you click"y", it adds all new packages which are incompatible with matplotlib. This time, I didn't update and said no. I also didn't use a separate environment, I just started with the import block and I didn't get the matplotlib error. I am getting new errors though, so far it can't read the file because of:


OSError: [Errno 22] Unable to open file (unable to open file: name = 'C:\Users8042\ATL03_20230203162831_06891807_006_01.h5', errno = 22, error message = 'Invalid argument', flags = 0, o_flags = 0)


I'll try to debug that.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
ok, so it looks like we have gone full circle here. Do you have a new clean environment using:

`conda create -n bathy python=3.10`

**Comment by slyoung20** [comment]
You can't create an environment; it updates Conda and gives you python 11 which conflicts with matplotlib. For some reason, Anaconda won't let me use Conda, without updating Conda which triggers Anaconda to update everything at once, python included. Then the matplotlib conflict is triggered. It wouldn't allow me to create an environment with a different version of python either, no matter how many times I did it. Finally, I just decided to go with Anaconda as it is with python 10. This works as long as I don't install or update anything.

I think the problem I have now is that my data is already in UTM. When I ran it with your file it worked. I'm going to try to download an IceSat file in WGS and see if that works. Fingers crossed, it should. Doing that now.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
Hi Nathan,

I checked one of the files in Arc, it is in WGS1984 projection. The EPSG should be 4326. Not sure why when I run the code I get:
# Find the epsg code
epsg_code = cshelph.convert_wgs_to_utm(latitude[0], longitude[0])
epsg_num = int(epsg_code.split(':')[-1])
print(f"epsg number: {epsg_num}")

epsg number: 32651

But when I run your file it works. Do I need to change any inputs that I may be missing?
Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
Hello Dr. Thomas,

OK, I got this working mostly now, looks like I'm missing one input: surface buffer. There are many, many, conflicts between all of the current packages that have to be resolved one by one. If you get somebody else with this issue, that's the problem. Honestly, I'm not even sure what the final setup is at this point, but it works.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Excellent, thanks for working all of this out. This isn't developed or tested on windows, so there's no guarantee that it can work. One way to check your environment credentials is:

`conda list`

As for the buffer argument, I updated the code recently so first uninstall cshelph.

`pip uninstall cshelph`

Then install it again:

`pip install cshelph`

If the error happens again, let me know and I'll update the code

**Comment by slyoung20** [comment]
Hi Nathan,

I got that fixed last night after I emailed you. It looks like you just set the buffer to -0.5 correct? I read that on the github page under your previous responses.  I'm getting this error now, everything works until I get here:

print('Fetching water temperature...')
water_temp = cshelph.get_water_temp(is2_h5_file, latitude, longitude)
print(f"water temp: {water_temp} deg C")


IndexError                                Traceback (most recent call last)
Cell In[20], line 2
      1 print('Fetching water temperature...')
----> 2 water_temp = cshelph.get_water_temp(is2_h5_file, latitude, longitude)
      3 print(f"water temp: {water_temp} deg C")

File ~\C-SHELPh-2.5.0\cshelph\cshelph.py:270, in get_water_temp(data_path, latitude, longitude)
    266 sea_temp_xr = xr.open_zarr(fsspec.get_mapper('s3://mur-sst/zarr', anon=True),consolidated=True)
    268 sea_temp = sea_temp_xr['analysed_sst'].sel(time=slice(start_date,end_date))
--> 270 sst = round(sea_temp.sel(lat=lat_med,lon=lon_med,method='nearest').load().values[0] - 273,2)
    272 return sst

IndexError: index 0 is out of bounds for axis 0 with size 0

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Ok, nearly there. Looks like it's having trouble getting the sea temp automatically. You can use the default value or have a quick look at the average sea temp for your area and pass it in using the water temp argument. In honesty the sea temp does impact the output much

**Comment by slyoung20** [comment]
Currently I see 3 codes for seatemp, I'm using the c-shelph 2.5 version right now. But if I redownload it from Github it has a different code.
What I have now:

def get_water_temp_redacted(data_path, latitude, longitude):
    '''
    pull down surface water temperature along the track from the JpL GHRSST opendap website.

    The GHRSST data are gridded tiles with dimension 17998 x 35999.
    To get the specific grid tile of the SST, you must convert from lat, lon coordinates
    to the gridded tile ratio of the SST data product using the coordinates of the IS2 data.
    '''
    # Get date from data filename
    file_date = data_path[-33:-25]
    year = file_date[0:4]
    month = file_date[4:6]
    day = file_date[6:8]
    julian_day = str(datetime.strptime(file_date, '%Y%m%d').timetuple().tm_yday)
    # Add zero in front of day of year string
    zero_julian_date = julian_day.zfill(3)

    # Calculate ratio of latitude from mid-point of IS2 track
    lat_avg = latitude.mean()
    lat_min = -90
    lat_max = 90
    scaled_lat_min = 0
    scaled_lat_max = 17998

    new_lat_ratio = round(((lat_avg - lat_min) / (lat_max - lat_min)) *
                    (scaled_lat_max - scaled_lat_min) + scaled_lat_min)

    # Calculate ratio of longitude from mid-point of IS2 track
    lon_avg = longitude.mean()
    lon_min = -180
    lon_max = 180
    scaled_lon_min = 0
    scaled_lon_max = 35999

    new_lon_ratio = round(((lon_avg - lon_min) / (lon_max - lon_min)) *
                    (scaled_lon_max - scaled_lon_min) + lon_min)

    # Access the SST data using the JpL OpenDap interface
    url = 'https://opendap.jpl.nasa.gov/opendap/OceanTemperature/ghrsst/data/GDS2/L4/GLOB/JpL/MUR/v4.1/'\
        + str(year) + '/' + str(julian_day) + '/' + str(file_date) \
        + '090000-JpL-L4_GHRSST-SSTfnd-MUR-GLOB-v02.0-fv04.1.nc'

    dataset = netCDF4.Dataset(url)

    # Access the data and convert the temperature from K to C
    sea_temp = dataset['analysed_sst'][0,new_lat_ratio, new_lon_ratio] - 273.15
    return sea_temp

def get_water_temp(data_path, latitude, longitude):
    '''
    pull down surface water temperature along the track from the MUR SST:
    https://aws.amazon.com/marketplace/pp/prodview-kvgy4vkuhavsc?sr=0-3&ref_=beagle&applicationId=AWSMpContessa#overview.
    '''
    # Get date from data filename
    file_date = data_path[-33:-25]

    start_date = str(datetime.strptime(file_date + ' 00:00:00', '%Y%m%d %H:%M:%S'))
    end_date = str(datetime.strptime(file_date + ' 23:59:59', '%Y%m%d %H:%M:%S'))

    # Calculate ratio of latitude from mid-point of IS2 track
    lat_med = np.median(latitude)

    # Calculate ratio of longitude from mid-point of IS2 track
    lon_med = np.median(longitude)

    sea_temp_xr = xr.open_zarr(fsspec.get_mapper('s3://mur-sst/zarr', anon=True),consolidated=True)

    sea_temp = sea_temp_xr['analysed_sst'].sel(time=slice(start_date,end_date))

    sst = round(sea_temp.sel(lat=lat_med,lon=lon_med,method='nearest').load().values[0] - 273,2)

    return sst

If I download from github now it has:

def get_water_temp(data_path, latitude, longitude):

    try:
        auth = earthaccess.login(strategy="netrc")
    except:
        print("Login credentials not found. Sign in manually, your netrc file will be created for next time")
        auth = earthaccess.login(strategy="interactive", persist=True)

    # Get date from data filename
    file_date = data_path[-33:-25]

    date_range = file_date[0:4] + '-' + file_date[4:6] + '-' + file_date[6:]

    location_df = pd.DataFrame({'longitude':longitude,'latitude':latitude})

    location_df = location_df.dropna(axis=0)

    minx, miny, maxx, maxy = np.min(location_df['longitude']), np.min(location_df['latitude']), np.max(location_df['longitude']), np.max(location_df['latitude'])

    lat_med = np.median(location_df['latitude'])
    lon_med = np.median(location_df['longitude'])

    Query = earthaccess.collection_query()

    # Use chain methods to customize our query
    collections = Query.keyword('GHRSST Level 4 CMC0.1deg Global Foundation Sea Surface Temperature Analysis').bounding_box(minx,miny,maxx,maxy).temporal(date_range,date_range).get(10)

    short_name = collections[0]["umm"]["ShortName"]

    Query = earthaccess.granule_query().short_name(short_name).version("3.0").bounding_box(minx,miny,maxx,maxy).temporal("2020-01-01","2020-01-01")

    granules = Query.get(10)

    ds_L3 = xr.open_mfdataset(earthaccess.open(granules),combine='nested',concat_dim='time',coords='minimal')

    sea_temp = ds_L3['analysed_sst'].sel(lat=lat_med,lon=lon_med,method='nearest').load()

    sst = round(np.nanmedian(sea_temp.values)-273,2)

    return sst

Which version is better to use?

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
That code should be in both. The first one is called redacted, which means it is no longer in use. It is not used in the code only the second one is used :)

**Comment by slyoung20** [comment]
Yes, I know that. The other one is this one:
def get_water_temp(data_path, latitude, longitude):

    try:
        auth = earthaccess.login(strategy="netrc")
    except:
        print("Login credentials not found. Sign in manually, your netrc file will be created for next time")
        auth = earthaccess.login(strategy="interactive", persist=True)

    # Get date from data filename
    file_date = data_path[-33:-25]

    date_range = file_date[0:4] + '-' + file_date[4:6] + '-' + file_date[6:]

    location_df = pd.DataFrame({'longitude':longitude,'latitude':latitude})

    location_df = location_df.dropna(axis=0)

    minx, miny, maxx, maxy = np.min(location_df['longitude']), np.min(location_df['latitude']), np.max(location_df['longitude']), np.max(location_df['latitude'])

    lat_med = np.median(location_df['latitude'])
    lon_med = np.median(location_df['longitude'])

    Query = earthaccess.collection_query()

    # Use chain methods to customize our query
    collections = Query.keyword('GHRSST Level 4 CMC0.1deg Global Foundation Sea Surface Temperature Analysis').bounding_box(minx,miny,maxx,maxy).temporal(date_range,date_range).get(10)

    short_name = collections[0]["umm"]["ShortName"]

    Query = earthaccess.granule_query().short_name(short_name).version("3.0").bounding_box(minx,miny,maxx,maxy).temporal("2020-01-01","2020-01-01")

    granules = Query.get(10)

    ds_L3 = xr.open_mfdataset(earthaccess.open(granules),combine='nested',concat_dim='time',coords='minimal')

    sea_temp = ds_L3['analysed_sst'].sel(lat=lat_med,lon=lon_med,method='nearest').load()

    sst = round(np.nanmedian(sea_temp.values)-273,2)

    return sst

Either one I try though still returns the same error.

Any ideas?

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
I noticed your original file path stated that the data was "processed". Did you select the layer you needed from the h5 file before running C-Shelph? Or what other processing did you have to do? I had a hard time selecting the layer I needed in ArcGIS from the h5 file.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
It says processed just because I only downloaded the data that I needed. The datafile you have is fine.

If you pass in the -wt argument along with an integer value, it should run. Have you tried this?

**Comment by slyoung20** [comment]
The problem is I need it to run for several areas with hundreds of files, so it would be best if it could automatically grab the SST. Let me try it just to see if I have other issues. I'm kind of concerned because it says the : index 0 is out of bounds for axis 0 with size 0, that makes me think that it isn't pulling any geographic information.

It's running again after I tried updating some of the packages. After that I'll try it with the SST passed in just to see what I get and let you know.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Yes, it's not funding any geographic data. I'm back in the office Tuesday so can look in detail then if you give me a filename. You can still run this hundreds of times and pass in -wt each time using a loop

**Comment by slyoung20** [comment]
Well, for different dates there will be different average temps. This is working so far, still running. I think I can work with it that way, minor inconvenience. I'll let you know what I get when it finishes.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
OK it ran, and completed. No more errors. This is the output from the second to last section. Is this what it should look like?
[cid:03e11e47-aff8-4336-91e5-41e36dc1c37d]
Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
Sherry, were you able to get cshelph working satisfactorily? The trickiest part now that you have the software running is tuning the parameters. Cshelph will not work well under all conditions. It is designed to perform best in clear shallow waters so is very study site dependent. The example in my notebook shows an ideal scenario. If there are no other questions related to this issue I will close it out. you can always ask any new questions under a new ticket

**Comment by slyoung20** [comment]
Hi Nathan,

Thank you. Yes, I did. Sorry for not getting back to you sooner. For now I'm manually entering the SST. I was wondering if you corrected the output to MLLW and if you did, what you used? I know for the US and US owned territories you can use VDatum, but outside the US I'm having a hard time finding a premade tool for conversion.

I have an idea I want to try for handling murky water. We'll see how that works.

Thank you so much for your assistance,

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by nmt28** [comment]
ok cool. ket me now how it goes. I do no MLLW correction. I transform the photons from ellipsoid (WGS84) to geoid (EGM2008) height, but that's it. Further corrections could be made by the user. Can you give me a filename you are using, I want to check out why the sea surface temp retrieval isn't working. Might take me a few days to get on to it though

**Comment by slyoung20** [comment]
Good afternoon Dr. Thomas,

I was wondering if you masked land before running the code? For example, most of the transects along Andros (Bahamas) go right over the island.

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________

**Comment by slyoung20** [comment]
Good afternoon Dr. Thomas,

Thank you for all your previous assistance. I've managed to get the code fully working and I'm cleaning the data in Arc. The last issue I'm having, is I can't seem to get start_lat and end_lat to work. What else do I need to add and where should I add it?

Thank you,

Sherry Young

PhD student, graduate instructor
Dept. of Geography and Geoinformation Sciences
George Mason University
4400 University Dr.
Fairfax, VA 22030

***@***.***
________________________________
